docs/development/design/privacy-levels.rst

```diff
@@ -0,0 +1,209 @@
+- If a project has a `404.html` file on the default version, it's served.
+- If a project has a ``robots.txt`` file on the default version, it's served.
+- A ``sitemap.xml`` file is always served.
+
+For the commercial site:
+
+- The project's dashboard is visible to only users that have read permission over the project.
+- The footer shows links to the project's dashboard (build, downloads, home) to only admin users.
+- Only versions with ``hidden = False`` are listed on the footer and appear on search results.
+- If a project has a `404.html` file on the default version, it's served if the user has permission over that version.
+- If a project has a ``robots.txt`` file on the default version, it's served if the user has permission over that version.
+- A ``sitemap.xml`` file is served if the user has permissions over the default version.
+
+Migration
+---------
+
+To differentiate between allowing or not privacy levels,
+we need to add a setting ``RTD_ALLOW_PRIVACY_LEVELS`` (``False`` by default).
+
+For the community and commercial site, we need to:
+
+- Remove/change code that depend on the project's privacy level.
+  Use the default version's privacy level instead.
+
+  - Display robots.txt
+  - Serve 404.html page
+  - Display sitemap.xml
+  - Querysets
+
+- Remove `Project.privacy_level` field
+- Migrate all protected versions to have the attribute ``hidden = True``,
+  and set their privacy level to public for the community site and private for the commercial site.
+- Change all querysets used to list versions on the footer and on search to use the ``hidden`` attribute.
+- Update docs
+
+For the community site:
+
+- Hide all privacy level related settings from the version form.
+- Don't expose privacy levels on API v3.
+- Mark all versions as public.
+
+For the commercial site:
+
+- Always hide the dashboard
+- Show links to the dashboard (downloads, builds, project home) on the footer only to admin users.
+
+Upgrade path overview
+---------------------
+
+Community site
+##############
+
+The default privacy level for the community site is public for versions and the dashboard is always public.
+
+Public project (community)
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Public version:
+  Normal use case, no changes required.
+- Protected version:
+  Users didn't want to list this version on the footer,
+  but also not deactivate it. This can be solved by using the new ``hidden`` setting.
+- Private version:
+  Users didn't want to show this version to their users yet or they were testing something.
+  This can be solved with the pull request builder feature and the ``hidden`` setting.
+
+Protected project (community)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Protected projects are not listed publicly.
+Probably users were hosting a WIP project,
+or personal public project.
+A public project with public versions should work for them,
+as we are removing listing all projects publicly (except for search).
```

Comment:
What is the migration path here? Change all protected projects to public projects with hidden versions?

Reply:
Yeah, I'm making more explicitly what migration to do here.

Reply:
Actually no, sorry. The migration path for the versions of protected and private projects follow the same rules as the public project. Since at the end what users were hidden is the docs built (versions).

The privacy level for project only controls if the dashboard should be public or not and if the project was listed or not.